unime/src/routes/credentials/[id]/+page.svelte

```diff
@@ -1,181 +1,47 @@
 
-  import { ActionSheet, Image, TopNavBar } from '$lib/components';
-  import { dispatch } from '$lib/dispatcher';
-  import { HeartStraightFillIcon, HeartStraightRegularIcon } from '$lib/icons';
-  import { state } from '$lib/stores';
-  import { getImageAsset } from '$lib/utils';
+  import { pageTitleStore, state } from '$lib/stores';
 
-  let credential = $state.credentials.find((c) => $page.params.id === c.id)!;
+  import CredentialHeader from './CredentialHeader.svelte';
+  import CredentialOverview from './CredentialOverview.svelte';
+  import DefaultRenderer from './DefaultRenderer.svelte';
+  import OpenBadgeRenderer from './OpenBadgeRenderer.svelte';
 
-  let title: string = credential.display_name;
+  // Credential cannot be loaded via load function since it's stored in the application state.
+  // TODO Credential should be loaded from backend via load function to handle invalid IDs properly.
+  function loadCredential() {
+    return $state.credentials.find((c) => $page.params.id === c.id);
+  }
 
-  let qrcodeText = JSON.stringify(credential, null, 0);
+  let credential: DisplayCredential | undefined = loadCredential();
 
-  let isFavorite: boolean = credential.metadata.is_favorite;
+  // The credential's metadata, e.g., the favorite status, may change.
+  // ESLint does not understand the reactive statement.
+  // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+  $: $state, (credential = loadCredential());
 
   $: {
-    const credential = $state.credentials.find((c) => $page.params.id === c.id)!;
-    // TODO: update title, isFavorite when changes in store
-    isFavorite = credential.metadata.is_favorite;
-    title = credential.display_name;
+    // Use `pageTitleStore` to get page title into layout.
+    pageTitleStore.set($LL.CREDENTIAL.NAVBAR_TITLE());
   }
-
-  const hiddenStandardFields: string[] = ['id'];
-  // TODO: custom metadata field related to NGDIL demo
-  const hiddenCustomFields: string[] = ['enrichment'];
-
-  const entries = { ...credential.data.credentialSubject };
-  hiddenStandardFields.concat(hiddenCustomFields).forEach((key) => delete entries[key]);
-
-  let credentialLogoUrl: string | null;
-
-  onMount(async () => {
-    credentialLogoUrl = await getImageAsset($page.params.id!);
-  });
 </script>
 
-<div class="content-height relative flex w-full flex-col">
-  <!-- Background (scaled, blurred, transparent) -->
-  <div class="absolute left-0 z-10 {credentialLogoUrl ? 'top-0 h-[270px]' : 'top-[50px] h-[220px]'} w-screen">
-    {#if !credentialLogoUrl}
-      <div class="relative h-[220px]"></div>
+{#if credential}
+  <div class="flex flex-col gap-7 px-4 pb-7">
+    <CredentialHeader {credential}>
+      <h1 class="text-center font-semibold">
+        {credential.data?.credentialSubject?.achievement?.name ?? credential.display_name}
+      </h1>
+    </CredentialHeader>
+    <CredentialOverview {credential} />
+    {#if credential.data?.credentialSubject?.achievement}
```

Comment:
We only want to use the `OpenBadgeRenderer` if the credential really _is_ an OpenBadge 3.0 (which is indicated by the "type" field), so we need to add another check here:
`&& credential.data?.type.includes('OpenBadgeCredential')`

Reply:
Saw you fixed this.